File: Problems/Algorithms/583. Delete Operation for Two Strings/delete_operation.rs
impl Solution {
    pub fn min_distance(word1: String, word2: String) -> i32 {
        let s1 = word1.as_bytes();
        let s2 = word2.as_bytes();
        let n1 = word1.len();
        let n2 = word2.len();
        let mut dp = vec![vec![0; n2+1]; n1+1];
        
        for (i, c1) in s1.iter().enumerate() {
            for (j, c2) in s2.iter().enumerate() {
                if c1 == c2 {
                    dp[i+1][j+1] = 1 + dp[i][j];
                } else {
                    dp[i+1][j+1] = dp[i][j+1].max(dp[i+1][j]);
                }
            }
        }
        
        (n1 + n2 - 2 * dp[n1][n2]) as i32
    }
}

File: Problems/Algorithms/583. Delete Operation for Two Strings/delete_operation.py
class Solution:
    def minDistance(self, word1: str, word2: str) -> int:
        n1, n2 = len(word1), len(word2)
        dp = [[0 for _ in range(n2+1)] for _ in range(n1+1)]

        for i, c1 in enumerate(word1):
            for j, c2 in enumerate(word2):
                if c1 == c2:
                    dp[i+1][j+1] = 1 + dp[i][j]
                else:
                    dp[i+1][j+1] = max(dp[i][j+1], dp[i+1][j])

        return n1 + n2 - 2 * dp[n1][n2]

